core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/BaseAwsJsonProtocolFactory.java

```diff
@@ -138,14 +149,15 @@ private <T> MetricCollectingHttpResponseHandler<T> timeUnmarshalling(HttpRespons
     }
 
     private StructuredJsonGenerator createGenerator(OperationInfo operationInfo) {
-        AwsJsonProtocol protocol = protocolMetadata.protocol();
-        if (operationInfo.hasPayloadMembers()
-            || protocol == AwsJsonProtocol.AWS_JSON
-            || protocol == AwsJsonProtocol.SMITHY_RPC_V2_CBOR) {
-            return createGenerator();
-        } else {
+        Boolean useNoOp = operationInfo.addtionalMetadata(USE_NO_OP_GENERATOR);
```

Comment:
Nit: `addtional` should be `additional`

Reply:
Yeah, the method is already published like that, see [here](https://github.com/aws/aws-sdk-java-v2/blob/4496c1387b4b937e7303425d32936cd7f095195c/core/protocols/protocol-core/src/main/java/software/amazon/awssdk/protocols/core/OperationInfo.java#L136). Not sure if it's worth adding an override with the correct spelling, let me know.

Reply:
fine